src/rgw/setup/rgw.py

```diff
@@ -296,48 +289,13 @@ def _create_rgw_user(conf: MappedConf):
                 return rc
 
     @staticmethod
-    def _create_symbolic_link_fid(client_instance_count: int, sysconfig_file_path: str):
-        """ Create symbolic link of FID sysconfig files."""
-        hare_generated_fid_files = Rgw._get_files(sysconfig_file_path + f'/{const.COMPONENT_NAME}-0x*')
-        count = len(hare_generated_fid_files)
-        Log.info(f'{const.COMPONENT_NAME} FID file count : {count}')
-        Log.info(f'Number of {const.COMPONENT_NAME} client instances - {client_instance_count}')
-        if count < client_instance_count:
-            raise SetupError(errno.EINVAL,
-                f'HARE-sysconfig file does not match {const.COMPONENT_NAME} client instances.')
-
-        # Create symbolic links of rgw-fid files created by hare.
-        # e.g rgw-0x7200000000000001\:0x9c -> rgw-1 , rgw-0x7200000000000001\:0x5b -> rgw-2
-        index = 1
-        for src_path in hare_generated_fid_files:
-            file_name = f'{const.COMPONENT_NAME}-' + str(index)      # e.g. rgw-1 for rgw file
-            dst_path = os.path.join(sysconfig_file_path, file_name)
-            Rgw._create_symbolic_link(src_path, dst_path)
-            index += 1
-
-    @staticmethod
-    def _create_symbolic_link(src_path: str, dst_path: str):
-        """create symbolic link."""
-        Log.debug(f'symbolic link source path: {src_path}')
-        Log.debug(f'symbolic link destination path: {dst_path}')
-        if os.path.exists(dst_path):
-            Log.debug('symbolic link is already present')
-            os.unlink(dst_path)
-            Log.debug('symbolic link is unlinked')
-        os.symlink(src_path, dst_path)
-        Log.info(f'symbolic link created successfully from {src_path} to {dst_path}')
-
-    @staticmethod
-    def _parse_endpoint_values(client_instance_file: str):
-        """Read sysconfig file generated by hare
-         1) Read symblink file '{client_instance_file}' as default endpoints in config phase.
-         2) fetch endpoint values for running radosgw-admin tool.
-        """
-        endpoints = {}
-        with open(client_instance_file) as ep_file:
-            for line in ep_file:
-                ep_name, ep_value = line.partition('=')[::2]
-                endpoints[ep_name.strip()] = str(ep_value.strip())
+    def _parse_endpoint_values(conf: MappedConf):
+        """Fetch endpoint values from hctl fetch-fids for running radosgw_admin tool."""
+        hare_config_dir = Rgw._get_hare_config_path(conf)
+        fetch_fids_cmd = f'hctl fetch-fids -c {hare_config_dir}'
```

Comment:
we need to find a way to get unique fid endpoint value for each rgw instance.
if two rgw instance gets same motr & hare endpoint value using 'hctl fetch fid' command then RGW start up script will always fail wit MotrInit() calls since both service may end up having single same endpoint.

Reply:
Each server pod will have unique fid endpoint value(machine-ids are different for each server pod)
[root@cortx-server-headless-svc-ssc-vm-g4-rhev4-0910-2 /]# hctl fetch-fids -c /etc/cortx/hare/config/cbdbeacbea2e40b88014253831feddfe/
[
  {
    "name": "hax",
    "fid": "0x7200000000000001:0x91",
    "ep": "inet:tcp:cortx-server-headless-svc-ssc-vm-g4-rhev4-0910-2@22001"
  },
  {
    "name": "rgw",
    "fid": "0x7200000000000001:0x94",
    "ep": "inet:tcp:cortx-server-headless-svc-ssc-vm-g4-rhev4-0910-2@21501",
    "ha_ep": "inet:tcp:cortx-server-headless-svc-ssc-vm-g4-rhev4-0910-2@22001",
    "profile_fid": "0x7000000000000001:0xea"
  }
]
[root@cortx-server-headless-svc-ssc-vm-g4-rhev4-0910-2 /]# exit
[root@ssc-vm-g4-rhev4-0910 ~]# kubectl exec -it cortx-server-ssc-vm-g4-rhev4-0910-1-6db8c95df5-fjzfp -c cortx-rgw -- /bin/bash
[root@cortx-server-headless-svc-ssc-vm-g4-rhev4-0910-1 /]# hctl fetch-fids -c /etc/cortx/hare/config/a35e2fbdf58241bc9386b3bbaf616b28/
[
  {
    "name": "hax",
    "fid": "0x7200000000000001:0x89",
    "ep": "inet:tcp:cortx-server-headless-svc-ssc-vm-g4-rhev4-0910-1@22001"
  },
  {
    "name": "rgw",
    "fid": "0x7200000000000001:0x8c",
    "ep": "inet:tcp:cortx-server-headless-svc-ssc-vm-g4-rhev4-0910-1@21501",
    "ha_ep": "inet:tcp:cortx-server-headless-svc-ssc-vm-g4-rhev4-0910-1@22001",
    "profile_fid": "0x7000000000000001:0xea"
  }
]

Reply:
As far as RGW instances are scaled with horizontal scaling of server pods, i believe we are good and that's the decided approach today. However if multiple RGW instances per Server pod is decided to be used later also, its better if we design our code thinking that in mind. So what Sachitanand mentioned above matters in the long run. For a given client name, like rgw-1, can we get all these corresponding FIDs, if yes, we should integrate that way

Reply:
i understand after our discussion. @sachitanands can resolve this comment if he agrees